### PR TITLE
feat: add abbreviated building names to meeting times properties

### LINKED ISSUE #287: Add abbreviated building names alongside the "where" attribute on MeetingTime attribute
```json
"meetingTimes": [
      {
        "type": "Every Week",
        "time": "12:30 pm - 1:20 pm",
        "days": "TWF",
        "where": "David Turpin Building A110",
        "dateRange": "Sep 08, 2021 - Dec 06, 2021",
        "scheduleType": "Lecture",
        "instructors": [
          "Laura Elizabeth Teshima (P)"
        ]
      }
    ]
```
Modify the endpoint for `sections` and add a `shortWhere` attribute (or something more appropriate) on the `MeetingTime` type. In this case it would be `"shortWhere": "DTB A110"`.

Start with modifying the `getSections` response and augment the raw data.
https://github.com/VikeLabs/courseup/blob/main/functions/src/sections/Section.service.ts#L11

### PR DESCRIPTION
**Pull request by aomi**
# Description
Parses the `where` attribute, splits it into its building and room respectively. Using the building, it looks it up in a map. If it is found then the room number, building long & short form are appended to the `MeetingTime` type.

Not sure if the regex is correct without doing some analysis on the entire dataset. That might be a later thing...z

Closes #287 

## Screenshots
image

## Checklist

- [ ] The code follows all style guidelines.
- [ ] The code passes all required tests.
- [ ] The code is documented.
- [ ] The code includes tests.
- [ ] I have self-reviewed my changes and have done QA.

### CODE REVIEW

**Comment by aomi** [comment]
Still need to generate the routes and such. Hopefully that doesn't break everything :|

File: functions/src/constants.ts [hunk 1]
```diff
@@ -10,3 +10,81 @@ export type Term =
   | '202205';
 
 // add some code?
```
Comment (keithradford) [inline functions/src/constants.ts:12]: is this a TODO?

File: functions/src/sections/Section.model.ts [hunk 2]
```diff
@@ -3,7 +3,22 @@ import {
...
+export type MeetingTimes = {
+  type: string;
+  time: string;
+  days: string;
+  where: string;
+  dateRange: string;
+  scheduleType: string;
+  instructors: string[];
+  building?: string;
+  buildingAccronym?: string;
```
Comment (keithradford) [inline functions/src/sections/Section.model.ts:15]:
```suggestion
  buildingAbbreviation?: string;
```
 I'm not sure `acronym` is the best here given in some cases it's just a shortened version.

**Review by keithradford — COMMENTED** [review]
Would be nice to have some test cases perhaps

File: functions/src/constants.ts (hunk 1 above)
Comment (aomi) [reply functions/src/constants.ts:12]: Uh that has always been there and unrelated to this...

Comment (keithradford) [reply functions/src/constants.ts:12]: ya I was just wondering

File: src/lib/fetchers.tsx [hunk 3]
```diff
@@ -73,17 +77,10 @@ export type LevelType = 'law' | 'undergraduate' | 'graduate';
...
-  dateRange: string;
-  scheduleType: string;
-  instructors: string[];
-}
-
-export interface ClassScheduleListing {
+/**
+ * From T, pick a set of properties whose keys are in the union K
+ */
+export interface PickClassScheduleListingExcludeKeyofClassScheduleListingMeetingTimes {
```
Comment (keithradford) [inline src/lib/fetchers.tsx:83]: this is funny lol

**Review by keithradford — APPROVED** [review]

**Pull request merged**